### DISCUSSION TITLE
Oxygen Distributor Multiblock

### DISCUSSION THREAD

**Discussion by ghost** [terrarium-earth/Ad-Astra#156] [Ideas]
I and a few other people wanted to build a big moon station, only problem is there is not enough blocks available with the normal Oxygen Distributor, so i had a idea of a Oxygen Distributor Multiblock than can support a very massive amount blocks (maybe something like 100K blocks or even more), and the Bigger (or smaller) you build it would support a Higher/Lower block count.

**Comment by AlexNijjar** [comment]
the problem with that is it would kill performance as the oxygen distributor would need to actively check 100k blocks. in the config, there’s an option to change the max supported blocks and the check frequency. for your specific scenario,. you can set that value to be very high and decrease the frequency significantly in order to accommodate for the size of the space station.

**Reply by PeaceIsBliss** [reply]
I too wanted to go big, and found this post and some reality kicked in. I made a station with what was hopefully going to be a greenhouse. The area is quite large and I shouldve thought this out more. Worst case scenario - forcing the server to deal with the area checks - is it possible to stack enough oxygen distributors to fill it with oxygen? Do they need spread out, can they only see a certain distance for checks? Also, how often does this check occur? I may be able to do some throttling with CC:Tweaked and Player detection, only allowing distributors to activate when a player is in a certain area. The more information I have the better.

**Reply by blagdorfinguy** [reply]
If you're looking for an easier fix with your situation, you could always cover the inaccessible areas of your greenhouse (im imagining a dome with a high ceiling) with a layer of barrier blocks, to "enclose" the bottom, walkable area
